src/mongoc/mongoc-util.c

```diff
@@ -123,16 +123,18 @@ bool
 _mongoc_lookup_bool (const bson_t *bson, const char *key, bool default_value)
 {
    bson_iter_t iter;
+   bson_iter_t child;
 
    if (!bson) {
       return default_value;
    }
 
-   if (!bson_iter_init_find (&iter, bson, key)) {
+   BSON_ASSERT (bson_iter_init (&iter, bson));
+   if (!bson_iter_find_descendant (&iter, key, &child)) {
```

Comment:
Thanks. I believe this previously worked when I was originally using `bson_lookup_bool()` and broke after I rebased and had to switch over to ` _mongoc_lookup_bool()` (due to https://github.com/mongodb/mongo-c-driver/commit/381689eee9b5b512eb2650b4ddd47bf97127dd1f#diff-4c820a241dabdb8a4c8b3b9bf488ca5b). I guess they weren't exactly drop-in replacements of each other.